src/agent/src/device.rs

```diff
@@ -717,20 +758,48 @@ fn split_vfio_pci_option(opt: &str) -> Option<(&str, &str)> {
     }
 }
 
+// VFIO device could be PCI or AP, which requires different handling. The first option is checked
+// for the prefix, `pci:` or `ap:` (both within one device are not possible).
+#[instrument]
+async fn vfio_device_handler(device: &Device, sandbox: &Arc<Mutex<Sandbox>>) -> Result<SpecUpdate> {
+    let first_option = device
+        .options
+        .get(0)
+        .ok_or_else(|| anyhow!("VFIO device {} had no options", device.container_path))?;
+    match first_option.split_once(':') {
+        Some(("pci", _)) => vfio_pci_device_handler(device, sandbox).await,
+        Some(("ap", _)) => vfio_ap_device_handler(device, sandbox).await,
+        _ => Err(anyhow!(
+            "VFIO device option {} did not indicate `pci:` or `ap:`",
+            first_option
+        )),
+    }
+}
+
 // device.options should have one entry for each PCI device in the VFIO group
 // Each option should have the form "DDDD:BB:DD.F=<pcipath>"
 //     DDDD:BB:DD.F is the device's PCI address in the host
 //     <pcipath> is a PCI path to the device in the guest (see pci.rs)
-async fn vfio_device_handler(device: &Device, sandbox: &Arc<Mutex<Sandbox>>) -> Result<SpecUpdate> {
+#[instrument]
+async fn vfio_pci_device_handler(
+    device: &Device,
+    sandbox: &Arc<Mutex<Sandbox>>,
+) -> Result<SpecUpdate> {
     let vfio_in_guest = device.field_type != DRIVER_VFIO_GK_TYPE;
     let mut pci_fixups = Vec::<(pci::Address, pci::Address)>::new();
     let mut group = None;
 
-    for opt in device.options.iter() {
+    for pci_opt in device.options.iter() {
+        let opt = pci_opt.strip_prefix("pci:").ok_or_else(|| {
+            anyhow!(
+                "VFIO PCI device option {} unexpectedly missed `pci:` indication",
+                pci_opt
+            )
+        })?;
         let (host, pcipath) =
-            split_vfio_option(opt).ok_or_else(|| anyhow!("Malformed VFIO option {:?}", opt))?;
-        let host =
-            pci::Address::from_str(host).context("Bad host PCI address in VFIO option {:?}")?;
+            split_vfio_pci_option(opt).ok_or_else(|| anyhow!("Malformed VFIO option {:?}", opt))?;
```

Comment:
Uh... doesn't the fact that you're just changing `split_vfio_option` here mean you broke compile wtih the earlier patch that renamed the function?